pyccel/codegen/printing/ccode.py

```diff
@@ -268,6 +393,84 @@ def function_signature(self, expr):
             arg_code = ', '.join('{0}{1}'.format(self.get_declare_type(i), i) for i in expr.arguments)
         return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_NumpyUfuncBase(self, expr):
+        """ convert a python expresion with a numpy function call to c
+        function call
+
+        Parameters:
+        -----------
+            expr (Pyccel ast node): python expresion with a numpy function call
+
+        Returns:
+        --------
+            string: equivalent of the expression in c language
+
+        Example:
+        -------
+            numpy.cos(x) ==> cos(x)
+
+        """
+        # add necessary include
+        self._additional_imports.add('math.h')
+        type_name = type(expr).__name__
+        func_name = numpy_ufunc_to_c_real[type_name]
```

Comment:
Please add a try/except statement to catch possible `KeyError`s if the user requests a function which is not implemented. This can be used to raise a neat `PYCCEL_RESTRICTION_TODO`

Reply:
done